ee/modules/502-delivery/crds/werfsource.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: werfsources.deckhouse.io
+  labels:
+    heritage: deckhouse
+    module: delivery
+spec:
+  group: deckhouse.io
+  scope: Cluster
+  names:
+    plural: werfsources
+    singular: werfsource
+    kind: WerfSource
+  preserveUnknownFields: false
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          description: |
+            Delivery source configuration for Argo CD and werf bundle. It is used to create a registry
+            configuration for Argo CD Image Updater and optionally to create a OCI Helm repository
+            in Argo CD to use with werf bundles.
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - imageRepo
+              properties:
+                imageRepo:
+                  type: string
+                  description: Image repository.
+                  x-example: registry.example.com/my-project
+                  pattern: ^(?:[a-z0-9]+(?:[.-][a-z0-9]+)*)(/(?:[a-z0-9._-]+))*$
+                apiURL:
+                  type: string
+                  pattern: ^https?://(?:[a-z0-9]+(?:[.-][a-z0-9]+)*)$
+                  description: |
+                    URL of the registry API. It is automatically derived from `imageRepo` if not specified.
+                    E.g. if `imageRepo=cr.example.com/myproject`, then `apiURL` is derived as
+                    `https://cr.example.com/`.
+                pullSecretName:
```

Comment:
Let's add a validating webhook for pullSecretName if provided as in the ingress-nginx module.
https://github.com/deckhouse/deckhouse/blob/1481737a218ea3546f2ee6a9effaf472040d0581/modules/402-ingress-nginx/webhooks/validating/ingressnginxcontroller#L60-L65

Reply:
Can be moved to Phase 2.

Reply:
Good, let's take it on the next iteration. For now, deckhouse hook will return errors if there is something wrong with the pull secret